tests/integration/tests/test_ingress.py

```diff
@@ -14,23 +15,35 @@
 def test_ingress(session_instance: List[harness.Instance]):
 
     util.stubbornly(retries=5, delay_s=2).on(session_instance).until(
-        lambda p: "cilium-ingress" in p.stdout.decode()
-    ).exec(["k8s", "kubectl", "get", "service", "-n", "kube-system", "-o", "json"])
+        lambda p: "ingress" in p.stdout.decode()
+    ).exec(["k8s", "kubectl", "get", "service", "-A", "-o", "json"])
 
     p = session_instance.exec(
         [
             "k8s",
             "kubectl",
             "get",
             "service",
-            "-n",
-            "kube-system",
-            "cilium-ingress",
-            "-o=jsonpath='{.spec.ports[?(@.name==\"http\")].nodePort}'",
+            "-A",
+            "-o=json",
         ],
         capture_output=True,
     )
-    ingress_http_port = p.stdout.decode().replace("'", "")
+
+    ingress_http_port = None
+    services = json.loads(p.stdout.decode())
+
+    ingress_services = [svc for svc in services["items"] if "ingress" in svc["metadata"]["name"]]
```

Comment:
this is dangerous and could very well lead in a future where this test passes due to false positives. We know the resource we have deployed, so we should check for that.

Reply:
What would you suggest here? I added this to check for *ingress or in the other test *my-gateway because cilium seems to add their name in front of the resources it creates... So I can't check for the exact resource name with contour and cilium...

Reply:
in that case, do not try to guess around implementation details. rather, start from the gateway object and track down the linked service.